Modules/_threadmodule.c

```diff
@@ -2354,6 +2363,108 @@ PyDoc_STRVAR(thread__get_main_thread_ident_doc,
 Internal only. Return a non-zero integer that uniquely identifies the main thread\n\
 of the main interpreter.");
 
+
+#ifdef HAVE_PTHREAD_GETNAME_NP
+/*[clinic input]
+_thread._get_name
+
+Get the name of the current thread.
+[clinic start generated code]*/
+
+static PyObject *
+_thread__get_name_impl(PyObject *module)
+/*[clinic end generated code: output=20026e7ee3da3dd7 input=35cec676833d04c8]*/
+{
+    // Linux and macOS are limited to respectively 16 and 64 bytes
+    char name[100];
+    size_t size = Py_ARRAY_LENGTH(name) - 1;
+    pthread_t thread = pthread_self();
+    int rc = pthread_getname_np(thread, name, size);
+    if (rc) {
+        errno = rc;
+        return PyErr_SetFromErrno(PyExc_OSError);
+    }
+
+    name[size] = 0;
+    return PyUnicode_DecodeFSDefault(name);
+}
+#endif  // HAVE_PTHREAD_GETNAME_NP
+
+
+#ifdef HAVE_PTHREAD_SETNAME_NP
+/*[clinic input]
+_thread.set_name
+
+    name as name_obj: object
+
+Set the name of the current thread.
+[clinic start generated code]*/
+
+static PyObject *
+_thread_set_name_impl(PyObject *module, PyObject *name_obj)
+/*[clinic end generated code: output=402b0c68e0c0daed input=b55d3d4279e2e831]*/
+{
+    if (!PyUnicode_Check(name_obj)) {
+        PyErr_Format(PyExc_TypeError, "expected str, got %T", name_obj);
+        return NULL;
+    }
+
+    const PyConfig *config = _Py_GetConfig();
+    char *encoding;
+    int res = _Py_EncodeUTF8Ex(config->filesystem_encoding, &encoding,
+                               NULL, NULL, 0, _Py_ERROR_STRICT);
+    if (res == -2) {
+        PyErr_Format(PyExc_RuntimeWarning,
```

Comment:
If it's a hard error it shouldn't be a Warning.
```suggestion
        PyErr_Format(PyExc_RuntimeError,
```

Reply:
I copied the code from encode_wstr_utf8(). Here is a fix https://github.com/python/cpython/pull/127420 for encode_wstr_utf8().